lib/pavilion/dir_db.py

```diff
@@ -92,24 +97,72 @@ def create_id_dir(id_dir: Path) -> (int, Path):
 
         return next_id, next_id_path
 
-
 def default_filter(_: Path) -> bool:
     """Pass every path."""
 
     return True
 
-
 def select(id_dir: Path,
-           filter_func: Callable[[Path], bool] = default_filter) -> List[Path]:
+           filter_func: Callable[[Path], bool] = default_filter,
+           fn_base: int = 10) -> List[Path]:
     """Return a list of all test paths in the given id_dir that pass the
     given filter function. The paths returned are guaranteed (within limits)
     to be an id directory, and only paths that pass the filter function
-    are returned."""
+    are returned.
+
+    :param id_dir: A directory containing subdirectories to be filtered and
+                   removed.
+    :param filter_func: The filter function used to filter subdirectories in
+                        id_dir.
+    :param fn_base: The Integer base used to check whether directory names are
+                    valid intergers or hex. 16 for hex, 10 (Default) for ints.
+    :return passed: The list of path objects that passed the filter.
+    """
 
     passed = []
     for path in id_dir.iterdir():
-        if path.name.isdigit() and path.is_dir():
+
+        # Check if dir_name is valid hex or int
+        # fn_base = 16 for hex, 10 for int
+        try:
+            int(path.name, fn_base)
+        except ValueError:
+            continue
+
+        if path.is_dir():
             if filter_func(path):
                 passed.append(path)
 
     return passed
+
+def delete(id_dir: Path,
+           filter_func: Callable[[Path], bool] = default_filter,
+           verbose: bool = False) -> int:
+    """Deletes a directory and it's entire contents.
+
+    :param id_dir: The directory to iterate through
+    :param verbose: True or False based on if you want verbose output. Default
+                    is false.
+    :return int count: The number of directories removed.
+    """
+
+    count = 0
+    error_list = []
+
+    lock_path = id_dir.with_suffix('.lock')
+    with lockfile.LockFile(lock_path):
+        for path in select(id_dir, filter_func):
+            try:
+                shutil.rmtree(path)
+            except OSError as err:
+                error_list.append("Could not remove {} {}: {}"
+                                  .format(id_dir.name, path, err))
+                continue
+            count += 1
+            if verbose:
+                output.fprint("Removed {} {}.".format(id_dir.name,
+                                                      path.name))
+
+    reset_pkey(id_dir)
+    return count, error_list
```

Comment:
Return a list of removed directory paths, and a list of errors. Get rid of all prints.